packages/server/parseFormBody.ts

```diff
@@ -24,26 +34,34 @@ const parseRes = (res: HttpResponse) => {
 }
 
 const parseFormBody = async ({res, contentType}: ParseFormBodySignature): Promise<JSON | null> => {
+  let parsedBody: unknown
+  const parsedUploadables: UploadableBufferMap = {}
   const resBuffer = await parseRes(res)
   if (!resBuffer) return null
   const parts = getParts(resBuffer, contentType) as MultipartField[]
   if (!parts) return null
 
   try {
-    let parsedBody
     parts.forEach(({name, data, type}) => {
       if (name === 'body') {
         parsedBody = JSON.parse(Buffer.from(data).toString())
-        return
+      } else if (name.startsWith('uploadables')) {
+        const [, key] = name.split('.')
+        parsedUploadables[key] = {
+          contentType: type,
+          buffer: Buffer.from(data)
+        } as UploadableBuffer
       }
-      if (!name.startsWith('uploadables')) return
-      const [, key] = name.split('.')
-      parsedBody.payload.variables[key] = {
-        contentType: type,
-        buffer: Buffer.from(data)
-      } as Uploadable
     })
-    return parsedBody
+    if (!parsedBody || typeof parsedBody !== 'object') return null
+    if (Object.keys(parsedUploadables).length) {
+      const validParsedBody = parsedBody as FetchHTTPData
+      validParsedBody.payload.variables = {
+        ...validParsedBody.payload.variables,
+        ...parsedUploadables
+      }
+    }
+    return parsedBody as JSON
```

Comment:
after validation, we know the type is going to be a superset of `FetchHTTPData`, so we can use that since it's more specific than JSON